File: src/test/java/javabot/operations/InfoOperationTest.java
package javabot.operations;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

import com.google.inject.Inject;
import javabot.BaseMessagingTest;
import javabot.dao.FactoidDao;
import javabot.model.Factoid;
import org.testng.annotations.Test;

public class InfoOperationTest extends BaseMessagingTest {
  @Inject
  private FactoidDao factoidDao;

  @Test
  public void info() {
    Factoid factoid = null;
    String key = "whatwhat";
    String value = "ah, yeah";
    String user = "test";
    try {
      LocalDateTime now = LocalDateTime.now();
      factoid = factoidDao.addFactoid(user, key, value);
      factoid.setUpdated(now);
      factoidDao.save(factoid);
      testMessage("~info " + key, String.format("%s was added by: %s on %s and has a literal value of: %s", key, user,
          now.format(DateTimeFormatter.ofPattern(InfoOperation.INFO_DATE_FORMAT)), value));
    } finally {
      factoid = factoidDao.getFactoid(key);
      if (factoid != null) {
        factoidDao.delete(factoid);
      }
    }
  }
}

File: src/test/kotlin/javabot/operations/InfoOperationTest.kt
package javabot.operations

import com.google.inject.Inject
import javabot.BaseTest
import javabot.dao.FactoidDao
import javabot.dao.LogsDaoTest
import org.testng.Assert
import org.testng.annotations.Test
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

class InfoOperationTest : BaseTest() {
    @Inject
    protected lateinit var factoidDao: FactoidDao
    @Inject
    protected lateinit var operation: InfoOperation
    @Inject
    protected lateinit var factoidOperation: GetFactoidOperation

    @Test fun info() {
        val key = "whatwhat"
        val value = "ah, yeah"
        val user = "test"
        try {
            val now = LocalDateTime.of(2014,3,23,21,12)
            factoidDao.addFactoid(user, key, value, LogsDaoTest.CHANNEL_NAME,  now)
            val format = now.format(DateTimeFormatter.ofPattern(InfoOperation.INFO_DATE_FORMAT))
            var response = operation.handleMessage(message("info " + key))
            Assert.assertEquals(response[0].value, "${key} was added by: ${user} on ${format} and has a literal value of: ${value}")
            response=factoidOperation.handleMessage(message("whatwhat"))
            Assert.assertEquals(response[0].value, "botuser, whatwhat is ah, yeah")
            response = operation.handleMessage(message("info " + key))
            Assert.assertEquals(response[0].value, "${key} was added by: ${user} on ${format} and has a literal value of: ${value}")
        } finally {
            var factoid = factoidDao.getFactoid(key)
            if (factoid != null) {
                factoidDao.delete(factoid)
            }
        }
    }
}
